control-plane/agents/core/src/core/states.rs

```diff
@@ -0,0 +1,88 @@
+use common_lib::types::v0::{
+    message_bus::{Nexus, NexusId, Pool, PoolId, Replica, ReplicaId},
+    store::{nexus::NexusState, pool::PoolState, replica::ReplicaState},
+};
+use std::{collections::HashMap, ops::Deref, sync::Arc};
+use tokio::sync::{Mutex, RwLock};
+
+/// Locked Resource States
+#[derive(Default, Clone, Debug)]
+pub(crate) struct ResourceStatesLocked(Arc<RwLock<ResourceStates>>);
+
+impl ResourceStatesLocked {
+    pub(crate) fn new() -> Self {
+        ResourceStatesLocked::default()
+    }
+}
+
+impl Deref for ResourceStatesLocked {
+    type Target = Arc<RwLock<ResourceStates>>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Resource States
+#[derive(Default, Debug)]
+pub(crate) struct ResourceStates {
+    /// Todo: Add runtime state information for nodes.
+    pub(crate) nexuses: HashMap<NexusId, Arc<Mutex<NexusState>>>,
+    pub(crate) pools: HashMap<PoolId, Arc<Mutex<PoolState>>>,
+    pub(crate) replicas: HashMap<ReplicaId, Arc<Mutex<ReplicaState>>>,
+}
+
+impl ResourceStates {
+    /// Update the pool states with the latest runtime information from Mayastor.
+    pub(crate) async fn update_pools(&mut self, pools: &[Pool]) {
+        for pool in pools {
+            let pool_state = PoolState::from(pool);
+            // If an entry exists for the pool, update it, otherwise add a new entry.
+            match self.pools.get(&pool.id) {
+                Some(locked_state) => {
+                    let mut state = locked_state.lock().await;
```

Comment:
some debug logs would be use full here? i.e debug!(?pool, "updating state");

Reply:
Added debug log.